packages/asset-swapper/src/utils/quote_requestor.ts

```diff
@@ -0,0 +1,104 @@
+    }
+    throw new Error(`Decoded asset data (${JSON.stringify(decodedAssetData)}) does not contain a token address`);
+}
+
+export class QuoteRequestor {
+    private readonly _rfqtMakerEndpoints: string[];
+    private readonly _schemaValidator: SchemaValidator = new SchemaValidator();
+
+    constructor(rfqtMakerEndpoints: string[]) {
+        this._rfqtMakerEndpoints = rfqtMakerEndpoints;
+    }
+
+    public async requestRfqtFirmQuotesAsync(
+        makerAssetData: string,
+        takerAssetData: string,
+        assetFillAmount: BigNumber,
+        marketOperation: MarketOperation,
+        takerApiKey: string,
+        takerAddress: string,
+        options?: Partial<RfqtFirmQuoteRequestOpts>,
+    ): Promise<SignedOrder[]> {
+        const { makerEndpointMaxResponseTimeMs } = _.merge({}, constants.DEFAULT_RFQT_FIRM_QUOTE_REQUEST_OPTS, options);
+
+        const buyToken = getTokenAddressOrThrow(makerAssetData);
+        const sellToken = getTokenAddressOrThrow(takerAssetData);
+
+        // create an array of promises for quote responses, using "undefined"
+        // as a placeholder for failed requests.
+
+        const responsesIfDefined: Array<undefined | AxiosResponse<SignedOrder>> = await Promise.all(
+            this._rfqtMakerEndpoints.map(async rfqtMakerEndpoint => {
+                try {
+                    return await Axios.get<SignedOrder>(`${rfqtMakerEndpoint}/quote`, {
+                        headers: { '0x-api-key': takerApiKey },
+                        params: {
+                            sellToken,
+                            buyToken,
+                            buyAmount: marketOperation === MarketOperation.Buy ? assetFillAmount.toString() : undefined,
+                            sellAmount:
+                                marketOperation === MarketOperation.Sell ? assetFillAmount.toString() : undefined,
+                            takerAddress,
+                        },
+                        timeout: makerEndpointMaxResponseTimeMs,
+                    });
+                } catch (err) {
+                    logUtils.warn(
+                        `Failed to get RFQ-T quote from market maker endpoint ${rfqtMakerEndpoint} for API key ${takerApiKey} for taker address ${takerAddress}`,
+                    );
+                    logUtils.warn(err);
+                    return undefined;
+                }
+            }),
+        );
+
+        const responses = responsesIfDefined.filter(
+            (respIfDefd): respIfDefd is AxiosResponse<SignedOrder> => respIfDefd !== undefined,
+        );
+
+        const ordersWithStringInts = responses.map(response => response.data); // not yet BigNumber
+
+        const validatedOrdersWithStringInts = ordersWithStringInts.filter(order =>
+            this._schemaValidator.isValid(order, schemas.orderSchema),
+        );
+
+        const orders: SignedOrder[] = validatedOrdersWithStringInts.map(orderWithStringInts => {
```

Comment:
What will happen if the MM returned a valid `SignedOrder` but for the wrong assets?  Seems like we should test for that and not include the order if it doesn't match what we expect wrt maker and taker assets.

Reply:
I did this here: https://github.com/0xProject/0x-monorepo/pull/2552